new/resources-anywhere.md

```diff
@@ -0,0 +1,211 @@
+
+To *legitimately* do this, you should use `chef_recipe_dsl` instead.
+
+#### `resources` conflicts between cookbooks
+
+Cookbooks will now be given an official namespace, `Chef::LoadedCookbooks::NAME`, to place things in.  Resources in `resources/x.rb` (and providers in ) will be created in that namespace to avoid these conflicts.  Constant access will remain the same (since superclasses are in the constant lookup chain).
+
+Accessing such resources directly via `Chef::Resource` is deprecated.  For backwards compatibility, these resources will also be aliased into `Chef::Resource` in Chef 12, with a deprecation warning for people who subclass or initialize from that location.
+
+#### `require 'chef/resource/file'`
+
+As the resources are moving, *requires* are also moving.  In Chef 12, we will emit a deprecation warning when you require 'chef/resource/x' instead of 'chef/resources/x', but it will still work.
+
+#### Redefining `dsl_name`
+
+Redefining `dsl_name` is deprecated and will not work in 13.  In Chef 12, the base `Chef::Resource` will include a `method_defined` hook that lets us detect this, cause the DSL to change, and emit a warning.
+
+#### Creating anonymous resource classes
+
+This procedure: `Chef::Resource::MyResource = Class.new(Chef::Resource) { ... }` could be used to create a resource currently and `my_resource` would work fine in recipes.  In Chef 13, that will no longer work, because we don't do anything magical for `Chef::Resource`.  In Chef 12, we will still check for `Chef::Resource::MyResource` when `method_missing(:my_resource)` is called, and if it is found, will return it and emit a deprecation warning.
+
+#### Directly calling `method_missing`
+
+Users who call `method_missing` directly (or override it and call super) with class names will break in Chef 13.  We will emit a deprecation warning in Chef 12 if `method_missing` is ever called for a method that is actually on the class.
+
+#### resource_name = 'something_else'
+
+Redefining the base DSL name in an LWRP subclass is deprecated and will not work with 13.  In 12, we catch it, emit a warning and change the DSL.
+
+### Risks To Specific Software
+
+While we're trying to maintain full compatibility, there's risk here, particularly for people doing metaprogramming of resources.  We will run the tests of the following packages, and make sure current versions
+still work (possibly emitting deprecation warnings):
+
+- chef-sugar
+- chefspec
+- chef-rewind
+- Poise
+- Crazytown
```

Comment:
can you add foodcritic? It digs into internals fairly heavily. CC @odcinek and @acrump